core/src/testFixtures/java/io/grpc/internal/AbstractTransportTest.java

```diff
@@ -919,7 +923,7 @@ public void basicStream() throws Exception {
     assertEquals("Hi. Who are you?", methodDescriptor.parseResponse(message));
     assertThat(clientStreamTracer1.nextInboundEvent())
         .matches("inboundMessageRead\\(0, -?[0-9]+, -?[0-9]+\\)");
-    if (sizesReported()) {
+    if (sizesReportedAtServer()) {
       assertThat(clientStreamTracer1.getInboundWireSize()).isGreaterThan(0L);
```

Comment:
I think we should add an assertion if (this instanceof InProcessTransportTest)) then assert that the outbound wire and uncompressed sizes are the same for the client stream tracer and the server stream tracer.

Reply:
Is it needed now as the new commit contains sizes being reported on both sides, client and server? 
Now all the transports will be having same checks. If required we can sure add this additional check for just Inprocess transport.